modules/native-ble/src/openbci_gui_native_helpers.cpp

```diff
@@ -0,0 +1,66 @@
+#include <ctype.h>
+#include <list>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+
+#include "SimpleBLE.h"
+#include "openbci_gui_native_helpers.h"
+#include "serialization.h"
+
+using json = nlohmann::json;
+
+#define SOFTWARE_REVISION_CHARACTERISTIC "00002a28-0000-1000-8000-00805f9b34fb"
+#define DEVICE_INFORMATION_SERVICE "0000180a-0000-1000-8000-00805f9b34fb"
+
+int scan_for_ganglions (int timeout_sec, char *output_json, int *output_len)
+{
+    std::list<SimpleBLE::Peripheral> peripherals;
+    auto adapter_list = SimpleBLE::Adapter::get_adapters ();
+    if (adapter_list.empty ())
+    {
+        return (int)GanglionScanExitCodes::PORT_OPEN_ERROR;
+    }
+
+    adapter_list[0].set_callback_on_scan_found (
+        [&peripherals] (SimpleBLE::Peripheral peripheral)
+        {
+            std::string identifier = peripheral.identifier ();
+
+            if (strncmp (identifier.c_str (), "Ganglion", 8) == 0 ||
+                strncmp (identifier.c_str (), "Simblee", 7) == 0)
```

Comment:
Why are we still checking for the name here?

Reply:
The algorithm goes something like this:
1. Scan for all devices
2. For each found device check if it's a Ganglion (which means it's identifier has either Ganglion or Simblee)
3. For each found Ganglion, temporarily connect and get the software revision string which describes the firmware version

On MacOS the device identifier is still Simblee - that hasn't changed. The fix is that we don't use the device identifier to determine the firmware version. We use the actual Software Revision Bluetooth characteristic advertised by the firmware. Note that we can't check that characteristic when using the dongle, so it's still important to have the firmware version in the advertised name for those connections. 

When using the dongle, I believe the identifier is reported correctly on MacOS (though this is worth double checking).